app/main.py

```diff
@@ -1,3 +1,98 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance | TypeError:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return Distance(self.km + other.km)
+        raise self.add_iadd_error_handler(other)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance | TypeError:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            self.km += other.km
+            return self
+        raise self.add_iadd_error_handler(other)
+
+    def __mul__(self, other: int | float) -> Distance | TypeError:
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return Distance(self.km * other.km)
+        raise self.mul_div_error_handler(other)
+
+    def __truediv__(self, other: int | float) -> Distance | TypeError:
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return Distance(round(self.km / other.km, 2))
+        raise self.mul_div_error_handler(other)
+
+    def __lt__(self, other: int | float | Distance) -> bool | TypeError:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return self.km < other.km
+        raise self.comparison_error_handler(other)
+
+    def __gt__(self, other: int | float | Distance) -> bool | TypeError:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return self.km > other.km
+        raise self.comparison_error_handler(other)
+
+    def __eq__(self, other: int | float | Distance) -> bool | TypeError:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return self.km == other.km
+        raise self.comparison_error_handler(other)
+
+    def __le__(
+        self, other: int | float | Distance
+    ) -> bool | callable(TypeError):
+        return not self > other
+
+    def __ge__(
+        self, other: int | float | Distance
+    ) -> bool | callable(TypeError):
+        return not self < other
+
+    @staticmethod
```

Comment:
1. Since your function doesn't do anything except raising an error with a message you can just raise TypeError in your code with a appropriate message.
2. If you want an error handler with certain logic it is define it in a separate class - no need to mix it with other logic. 
3. Also you have 3 methods that do exactly the same - so you could actually have only 1 with different parameters.
4. I understand that exceptions are beyond current scope of what you have learned - so you can remove this block of code and make it more simple.
5. Let me know if you have any questions